libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,324 @@
-was not always possible, so if you use any of the settings mentioned in the
-Breaking Changes section of the release notes, make sure you understand the
-alternatives that we provide.
+// REVIEWERS: What changes do users need to make to their config files? Will we
+// offer any kind of migration tooling to make this easier?
+
+{beats} 7.0 comes with several backwards incompatible configuration changes.
+Before upgrading, review the <<breaking-changes-7.0>> document. Where possible,
+we kept the old configuration options working, but deprecated them. However,
+deprecation was not always possible, so if you use any of the settings mentioned
+in the Breaking Changes section of the release notes, make sure you understand
+the alternatives that we provide.
+
+[[enable-ecs-compatibility]]
+==== Enable the compatibility layer for Elastic Common Schema (ECS) fields
+
+Starting with 7.0, the fields exported by {beats} conform to the
+{ecs-ref}/index.html[Elastic Common Schema (ECS)]. Many of the exported fields
+have been renamed. See {beats-ref}/breaking-changes-7.0.html[Breaking
+changes in 7.0] for the full list of changed names.
 
-===== modules.d configuration layout
+To help you transition to the new fields, we provide a compatibility layer in
+the form of ECS-compatible field aliases. To use the aliases, set
+`migration.6_to_7.enabled: true` in the Beat's configuration file *before* you
+run {beats} 7.0 to ingest data for the first time.
 
-Starting with the 6.0 version, Filebeat and Metricbeat are moving to a directory
-layout for configuration, where each module is configured in its own
-configuration file.
+// REVIEWERS: I'm not sure the following statement is correct. I was not able
+// to use the dashboards I saved from 6.x with 7.0. I did a very quick and dirty
+// upgrade of the stack tho so might have missed something critical. 
 
-While the all-in-one configuration is still fully supported, we recommend moving
-to the new layout at upgrade time. This typically means starting off with the new
-default configuration and modifying it with the custom settings that you had in
-your old files.
+The compatibility layer enables you to use 6.x dashboards and visualizations
+with indices created by {beats} 7.0 or later.
 
-// TODO: link to the docs on modules.d
+Did you run the Beat before reading this section? That's OK. See the clean-up
+steps described under <<missing-fields>>.
 
-===== New command `test config` command
+==== Upgrade dashboards
 
-Beats 6.0 introduces a new test command for testing the configuration file. For
-example:
+We recommend that you re-import the {kib} dashboards after upgrading {kib} and
+{beats}. This way, you can take advantage of the new {beats} dashboards created
+for the 7.0 release. 
+
+Re-importing the dashboards will overwrite any manual changes that you've
+made. To avoid overwriting existing dashboards, save any modified dashboards
+under a new ID before proceeding.
```

Comment:
We have some link/docs to show users how to do so? e.g.: https://www.elastic.co/guide/en/kibana/current/managing-saved-objects.html

Any modified object (visualizations, searches, ...) can be affected.